### PR TITLE
use dynamic plugin for side panel and action provider for rhoas

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5846

Task: https://issues.redhat.com/browse/ODC-6113

**Solution Description**: 

- converts Rhoas kafka connection side panel to use dynamic extensions
- converts Rhoas kafka connection  actions to use dynamic extensions

**Screen shots / Gifs for design review**: 

image

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/rhoas-plugin/console-extensions.json [hunk 1]
```diff
@@ -156,5 +156,58 @@
...
+  },
+  {
+    "type": "console.topology/details/tab-section",
+    "properties": {
+      "id": "rhoas-plugin-tab-section-resource",
+      "tab": "topology-side-bar-tab-resource",
+      "section": {
+        "$codeRef": "rhoasTopologyComponents.getResourceTabSectionForTopologySideBar"
+      }
+    }
```
Comment (sahil143) [inline frontend/packages/rhoas-plugin/console-extensions.json:211]: These should have `RHOAS_FLAG` as well

File: frontend/packages/rhoas-plugin/console-extensions.json [hunk 2]
```diff
@@ -156,5 +156,58 @@
...
+      "required": ["RHOAS_FLAG"]
+    }
+  },
+  {
+    "type": "console.topology/details/tab-section",
+    "properties": {
+      "id": "rhoas-plugin-tab-section-default",
+      "tab": "topology-side-bar-tab-details",
+      "section": {
+        "$codeRef": "rhoasTopologyComponents.getDetailsTabSectionForTopologySideBar"
```
Comment (sahil143) [inline frontend/packages/rhoas-plugin/console-extensions.json:199]: we can move these under `rhoasTopologySidebar` as well

File: frontend/packages/rhoas-plugin/console-extensions.json [hunk 3]
```diff
@@ -156,5 +156,58 @@
...
+      }
+    }
+  },
+  {
+    "type": "console.topology/details/tab-section",
+    "properties": {
+      "id": "rhoas-plugin-tab-section-resource",
+      "tab": "topology-side-bar-tab-resource",
+      "section": {
+        "$codeRef": "rhoasTopologyComponents.getResourceTabSectionForTopologySideBar"
```
Comment (sahil143) [inline frontend/packages/rhoas-plugin/console-extensions.json:209]: same as above

File: frontend/packages/rhoas-plugin/src/topology/components/kafka-connection-tab-section.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { GraphElement } from '@patternfly/react-topology';
+import { getResource } from '@console/topology/src/utils';
+import { KafkaConnection } from '../../utils/rhoas-types';
+import { TYPE_MANAGED_KAFKA_CONNECTION } from './const';
+import { DetailsComponent } from './DetailsComponent';
+import { ResourcesComponent } from './ResourceComponent';
+
+export const getDetailsTabSectionForTopologySideBar = (element: GraphElement) => {
+  if (element.getType() !== TYPE_MANAGED_KAFKA_CONNECTION) return undefined;
+  const resource = getResource(element) as KafkaConnection;
```
Comment (sahil143) [inline frontend/packages/rhoas-plugin/src/topology/components/kafka-connection-tab-section.tsx:11]:
```suggestion
  const resource = getResource<KafkaConnection>(element);
```

File: frontend/packages/rhoas-plugin/src/topology/components/kafka-connection-tab-section.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  return <DetailsComponent obj={resource} />;
+};
+
+export const getResourceTabSectionForTopologySideBar = (element: GraphElement) => {
+  if (element.getType() !== TYPE_MANAGED_KAFKA_CONNECTION) return undefined;
+  const resource = getResource(element) as KafkaConnection;
```
Comment (sahil143) [inline frontend/packages/rhoas-plugin/src/topology/components/kafka-connection-tab-section.tsx:17]: same as above

**Comment by wtrocki** [comment]
Verified with Operator running! Additional verification needed to check if plugin is loaded without operator/flag missing etc.

**Review by sahil143 — CHANGES_REQUESTED** [review]
@invincibleJai alert is showing after sidepanel is closed and opened again
Kapture 2021-09-28 at 12 06 45

**Comment by sahil143** [comment]
getting i18n error in console
image

**Review by sahil143 — APPROVED** [review]
/lgtm
/label qe-approved

verified locally, works as expected

**Comment by invincibleJai** [comment]
Propagate from epic:
/label docs-approved
/label px-approved

**Pull request merged**